### DISCUSSION TITLE
Hi! Just some newbie questions

### DISCUSSION THREAD

**Discussion by TimyIsCool** [navidrome/navidrome#3355] [Q&A]
I currently run my own Jellyfin server, but I've noticed some limitations when it comes to music management and playback. That's how I found Navidrome, and I have a couple of questions:

App Compatibility and User Experience: I've explored the demo of Navidrome's web UI and am quite impressed. However, I'm looking for an Android app that not only supports Navidrome but also offers a user experience similar to Spotify or YouTube Music. I've checked out DSub, but I'm curious if there are other apps I might have overlooked that provide that polished, mainstream streaming service feel?
Music Organization: Regarding the organization of my music library, can I structure my files so that each artist has their own folder, with subfolders for albums, or does Navidrome require all tracks to be in a single directory?

**Comment by Sevinfolds** [comment]
I use Tempo, previously used Subtracks. There is also Symfonium.
Of course you can structure it, that's how it's usually done. In Tempo you can turn on the display of music catalogs, they will just repeat your structure.

**Comment by deluan** [comment]
Apps: Check the ones that were tested with Navidrome here: https://www.navidrome.org/docs/overview/#apps Try some of them and see which one matches your preferences

Music Organization: Navidrome tries to abstain from dictating a file structure, it is tag-first (and only, for the most part), but following the pattern you suggested is what most people do. You can hit some issues with cover/artist jpg images depending on the file structure, you can read about it here: https://www.navidrome.org/docs/usage/artwork/

**Reply by TimyIsCool** [reply]
so with it being tag based system, how would the folder look like? and with artist cover images, does it simply extract it from the .mp3?

**Reply by deluan** [reply]
Being tag-based means it does not care for the folder structure, as it does not derive artist, album and track names from file paths. You can even put your whole library in just one folder, and as long as the files are tagged properly, Navidrome will organize your library correctly.

> does it simply extract it from the .mp3?

No it does not. Please read https://www.navidrome.org/docs/usage/artwork/
